src/oauth-client.ts

```diff
@@ -4,6 +4,10 @@ export interface Headers {
   [name: string]: string;
 }
 
+export interface OauthClientOptions {
+  redirectUri?: string;
```

Comment:
I think this might be safer if we require it to be a [`URL`](https://developer.mozilla.org/en-US/docs/Web/API/URL/URL) object, rather than a string.

[RFC 6749 3.1.2](https://datatracker.ietf.org/doc/html/rfc6749#section-3.1.2) requires that the Redirection URI must not have any fragments. It may have a query string (which the current behavior doesn't preserve).

Reply:
I'm fine to refactor it that way, but is it out of scope of this PR? It will involve refactoring the `redirectUrl` member inside oauth-facade, which has always been a string.

Reply:
I think we can do this validation just at the `OauthClient` layer.

`OauthFacade` is more of an internal interface, so I think it's ok to get a little closer to how the parameters will eventually be encoded and farther away from the most semantically appropriate representation.